### PR TITLE
#109 codefresh dev and production deployment generation implemented

### LINKED ISSUE #109: Add codefresh tagging/pushing step and production deployment
Add to the a new parallel step "publish" to the [codefresh build](https://codefresh.io/docs/docs/codefresh-yaml/steps/push/) that pushes the images tagged as "latest" and with a custom tag specified as a variable in the build.

Also make the `harness-codefresh` script generate 2 files, one for build-development and one for production.
The production script can be parametrized with a tag variable and will deploy a previously built deployment.

Another related change: use the parent's application name in docker image repository so that different (potentially overridden) builds of cloudharness images won't conflict.

### PR DESCRIPTION
**Pull request by filippomc**
Closes #109 

After this pr, the codefresh pipeline must be changed to point to either 
- deployment/codefresh-dev.yaml
- deployment/codefresh-prod.yaml

Also check your application deployment-configuration: codefresh-template.yaml splits into 

- codefresh-template-dev.yaml
- codefresh-template-prod.yaml

The content must be adapted from the content inside codefresh-template-dev.yaml and the current codefresh-template.yaml

### CODE REVIEW

File: utilities/cloudharness_utilities/codefresh.py [hunk 1]
```diff
@@ -45,48 +55,71 @@ def codefresh_build_step_from_base_path(base_path, build_step, fixed_context=Non
...
 
-def codefresh_build_spec(**kwargs):
+def codefresh_template_spec(template_path, **kwargs):
     """
     Create Codefresh build specification
     :return:
     """
 
-    build = get_template(CODEFRESH_BUILD_PATH)
+    build = get_template(template_path)
```
Comment (zsinnema) [inline utilities/cloudharness_utilities/codefresh.py:93]: I would do here ```return get_template(template_path).update(kwargs)```

Comment (filippomc) [reply utilities/cloudharness_utilities/codefresh.py:93]: update is a mutating method and returns null.

Comment (filippomc) [reply utilities/cloudharness_utilities/codefresh.py:93]: update is a mutating method and returns None

Comment (zsinnema) [reply utilities/cloudharness_utilities/codefresh.py:93]: hahah yeah you're right (shame)

**Pull request merged**